### PR TITLE
Make Screenshots deal with state directly

### PR DESCRIPTION
**Pull request by Pike**

### CODE REVIEW

File: frontend/src/modules/entitydetails/components/Screenshots.test.js [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
-        );
-
+        const wrapper = mountComponentWithStore(Screenshots, store, {
+            locale: 'kg',
+            source,
+        });
         wrapper.find('img').simulate('click');
 
-        expect(openLightboxFake.calledOnce).toEqual(true);
+        const newState = store.getState()[lightbox.NAME];
```
Comment (Pike) [inline frontend/src/modules/entitydetails/components/Screenshots.test.js:16]: I'm not too fond of the test here.

In a perfect world, I'd love the test here to pair with the test in https://github.com/mozilla/pontoon/blob/f000a47d7ff18cdae1c84b306711d34816f1db52/frontend/src/core/lightbox/components/Lightbox.test.js#L34-L45, i.e., catch the call to `dispatch`.

Mocking `store.dispatch` didn't seem to do that, much to my surprise.

Comment (Pike) [reply frontend/src/modules/entitydetails/components/Screenshots.test.js]: Figured it out, had to mock before mounting.

File: frontend/src/modules/entitydetails/components/Screenshots.tsx [hunk 2]
```diff
@@ -15,37 +16,26 @@ type Props = {
...
+        return null;
+    }
+
+    return (
+        <div className='screenshots'>
+            {images.map((urlWithLocale, i) => (
                 <img
                     src={urlWithLocale}
                     alt=''
                     key={i}
```
Comment (eemeli) [inline frontend/src/modules/entitydetails/components/Screenshots.tsx:35]:
```suggestion
            {images.map((urlWithLocale) => (
                <img
                    src={urlWithLocale}
                    alt=''
                    key={urlWithLocale}
```

Nitpick: Am I right in presuming that `urlWithLocale` will not be repeated within `images`? If so, it's a much better unique identifer than `i`.

**Review by eemeli — APPROVED** [review]
The functional code looks _much_ better.

File: frontend/src/modules/entitydetails/components/Screenshots.tsx (hunk 2 above)
Comment (Pike) [reply frontend/src/modules/entitydetails/components/Screenshots.tsx:35]: I agree that the index is probably the worst-practice choice here :-/

I guess it's safe to assume that folks are not going to have a comment with multiple references to the same image in there.

**Pull request merged**